tests/test_client.py

```diff
@@ -290,3 +293,52 @@ async def test_e2e_consume_multiple_topics():
         assert topic_2.total_events == total_events
 
         assert TopicManager.all_messages_consumed()
+
+
+@pytest.mark.asyncio
+async def test_streams_consume_events_with_initial_offsets(stream_engine: StreamEngine):
+    client = TestStreamClient(stream_engine)
+    event1 = b'{"message": "Hello world1!"}'
+    event2 = b'{"message": "Hello world2!"}'
+    process = Mock()
+
+    async with client:
+        await client.send(topic, value=event1, partition=0)
+        await client.send(topic, value=event1, partition=0)
+        await client.send(topic, value=event2, partition=1)
+
+        async def func_stream(consumer: Stream):
+            async for cr in consumer:
+                process(cr.value)
+
+        stream: Stream = Stream(
+            topics=topic,
+            consumer_class=TestConsumer,
+            name="my-stream",
+            func=func_stream,
+            initial_offsets=[
+                # initial topic offset is -1
+                TopicPartitionOffset(topic=topic, partition=0, offset=0),
```

Comment:
I understand initial_offset to indicate the offset of the first message to consume. so if you make the offsets 0, you would get all the messages.

I argue to make the offset for partition0 to be 1, and then confirm you get 1 message